logrotate.c

```diff
@@ -510,8 +510,13 @@ static int createOutputFile(char *fileName, int flags, struct stat *sb,
     if (fchmod(fd, (S_IRUSR | S_IWUSR) & sb->st_mode)) {
 	message(MESS_ERROR, "error setting mode of %s: %s\n",
 		fileName, strerror(errno));
-	close(fd);
-	return -1;
+
+		if (geteuid() == 0) {
+			/* If run as non-root, we may not be able to duplicate
+			the ownership of the file. Warn but don't fail. */
```

Comment:
Just had a look at this patch's history in the debian package and the check has been in the wrong block for about 5 years and nobody seems to have noticed. So it's probably not worth to add a libcap dependency for this. Will remove this one from this pr.